tardis/plugins/prometheusmonitoring.py

```diff
@@ -0,0 +1,73 @@
+from ..configuration.configuration import Configuration
+from ..interfaces.plugin import Plugin
+from ..interfaces.state import State
+from ..interfaces.siteadapter import ResourceStatus
+from ..utilities.attributedict import AttributeDict
+
+import logging
+from prometheus_client import Gauge, start_http_server
+
+
+class PrometheusMonitoring(Plugin):
+    """
+    The :py:class:`~tardis.plugins.prometheusmonitoring.PrometheusMonitoring`
+    implements an interface to monitor the state of the Drones using Prometheus.
+    """
+
+    def __init__(self):
+        self.logger = logging.getLogger("prometheusmonitoring")
+        self.logger.setLevel(logging.DEBUG)
+        config = Configuration().Plugins.PrometheusMonitoring
+
+        self._drones = {}
+
+        self._booting = Gauge("booting", "Booting drones")
+        self._running = Gauge("running", "Running drones")
+        self._stopped = Gauge("stopped", "Stopped drones")
+        self._deleted = Gauge("deleted", "Deleted drones")
+        self._error = Gauge("error", "Drones in error state")
+
+        start_http_server(config.port)
+
+    def compute_metrics(self) -> None:
+        booting = 0
+        running = 0
+        stopped = 0
+        error = 0
+        for drone in self._drones.values():
```

Comment:
Is it necessary to loop over all drones, when updating only the status of one? 
You know the previous state from `self._drones` and you know the current state, so you could use directly the `inc()` and `dec()` of the corresponding `Gauge`.

Reply:
Fixed in 339c3dd